contentcuration/kolibri_public/management/commands/rectify_incorrect_contentnode_source_fields.py

```diff
@@ -0,0 +1,162 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db.models import Exists
+from django.db.models import F
+from django.db.models import OuterRef
+from django.db.models import Q
+from django.db.models import Value
+from django.db.models.functions import Coalesce
+from django.utils import timezone
+from django_cte import With
+
+from contentcuration.models import Channel
+from contentcuration.models import ContentNode
+from contentcuration.models import User
+from contentcuration.utils.publish import publish_channel
+
+logger = logging.getLogger(__file__)
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+
+        parser.add_argument(
+            '--is_test',
+            action='store_true',
+            help="Indicate if the command is running in a test environment.",
+        )
+
+        parser.add_argument(
+            '--user_id',
+            type=int,
+            help="User ID for the operation",
+            required=True
+        )
+
+    def handle(self, *args, **options):
+        # Filter Date : July 9, 2023
+        # Link https://github.com/learningequality/studio/pull/4189
+        # The PR date for the frontend change is July 10, 2023
+        # we would set the filter day one day back just to be sure
+
+        is_test = options['is_test']
+        user_id = options['user_id']
+
+        if not is_test:
+            user_id = User.objects.filter(email='channeladmin@learningequality.org').values('pk')
```

Comment:
Here `user_id` will be equal to a dict with a `pk` key - need to key into it to get the user_id as a value by itself.